wbobeirne/joule-extension
# Save approved and rejected enables, redesign authorization

Closes #11.

### What This Does

Adds enabled and rejected domains to redux, and auto accepts or auto rejects on enable call if the domain is in redux. You can view and adjust any enabled or rejected sites in the settings menu.

Also does some visual touch-ups on the authorize prompt while in the neighborhood.

### Steps to Test

1. Go to any page and call `window.webln.enable()`, confirm you get prompt and it looks fly
2. Accept the enable request, and refresh the page
3. Repeat step 1 again, confirm it auto-enabled
4. Go to settings and confirm you see the domain
5. Remove the domain, and repeat step 1, confirm it pops up again
6. This time, reject the request (click button, not close prompt)
7. Repeat step 1, confirm it auto-rejects
8. Repeat steps 4 & 5
9. This time, uncheck "remember this decision" and confirm that subsequent tries still open the prompt

## Screenshots

### New Prompt

<img width="512" alt="screen shot 2018-12-03 at 1 02 34 am" src="https://user-images.githubusercontent.com/649992/49355791-253c3f00-f697-11e8-8eb5-da1fa8f1a26d.png">


### Settings

<img width="710" alt="screen shot 2018-12-03 at 12 07 24 am" src="https://user-images.githubusercontent.com/649992/49355779-15bcf600-f697-11e8-90e3-c9b87c96b569.png">
